### DISCUSSION TITLE
Help with "Could not resolve hostname" errors but DNS is fine

### DISCUSSION THREAD

**Discussion by lnlyssg** [nzbgetcom/nzbget#205] [Q&A]
I've just started getting a ton of messages like "Could not resolve hostname news.eweka.nl: Error -3 - Try again" for every server I have defined but I can't see why - I'm running from Docker and I can ping the addresses fine from the container e.g.

<img width="473" alt="Screenshot 2024-03-22 at 14 52 25" src="https://github.com/nzbgetcom/nzbget/assets/10902647/f13c0e7b-04cd-4cc4-9776-ffcba6aba8dd">

I have also run a test connection OK from the servers page so I'm struggling to work out what the issue could be.

**Comment by luckedea** [comment]
@lnlyssg Can you clarify:
1. Does this happen only on eweka or on other news servers too? How consistent is this?
2. What's the connection limit for that news server configuration?
3. Can you give us all the environment versions - docker, docker-compose, host system, nzbget version?

**Reply by lnlyssg** [reply]
1. It was every server I had defined, I was able to ping/dig all of them from the container but all servers gave the same error. I restarted the container multiple times but each time it came back with the same issues. Strangely enough this issue corrected itself later that day and hasn't reoccured since as far as I can see. I believe there were issues for around 5 hours in all.
2. Eweka was set to 30, I tried changing it lower and higher but this made no difference.
3. Docker version 20.10.23, build 876964a, using docker-compose on a Synology DS923+, nzbget v23.0

**Reply by luckedea** [reply]
@lnlyssg 

4. Does reboot make any difference?

**Reply by lnlyssg** [reply]
All my other containers, and the Synology itself, were all working fine so I was reluctant to reboot completely but if it happens again I will do so.

**Reply by noodlers** [reply]
I'm experiencing the same issue, except with the binary install on a Windows 10 system (version 23.0).

This happens for all configured servers. All configured servers "test" successful, however Messages tab filled with continuous "Could not resolve hostname: Error 11001 - No such host is known" error messages.

Servers can be pinged from host system, no DNS issues on host system with other applications.

No firewall changes have occurred.

Tried:
- Reloading from web interface
- Stopping and restarting from web interface
- Restarting host system multiple times

I'm not entirely sure what might have changed to cause this, but it has been ongoing for over a week without "fixing itself".

**Reply by luckedea** [reply]
Do you have any software that may impact this setup? VPN with SplitTunneling, AV software or some custom DNS tools (SmartDNS? SecureDNS?)

**Reply by noodlers** [reply]
No VPN software, or custom DNS tools that might influence the communication. Windows Defender is the only AV tool, and it has been tested both on and off with no change.

**After more investigation I noticed all my servers were set to IPv6 in the configuration**. **Changing these to IPv4 immediately worked**. I don't know why this is--the servers can all be pinged via IPv6 and I've confirmed IPv6 connectivity via other IPv6-only services. Also confirmed IPv6 DNS resolves.

**Reply by luckedea** [reply]
This is a good find which can explain many things. I'll do more tests about it, @phnzb please assist as well.

**Comment by paralovosca** [comment]
Same issue noticed today as well.  Using `nslookup` within `docker exec -ti nzbget /bin/sh` works fine. Yet, nzbget shows `[ERROR] Could not resolve hostname` for every server configured.

**Reply by paralovosca** [reply]
It's an intermittent issue but not sure what causes it. :(

**Comment by JohnnyGrey86** [comment]
Has there been any progress with this? I am also encountering this. nslookup resolves just fine. I am using Adguard Home running on my OPNsense router, which itself is using Unbound as the upstream DNS.

**Reply by dnzbk** [reply]
@JohnnyGrey86 Hi,
I'm investigating this issue - are you using IPv6? If so, does switching to IPv4 resolve the problem?

**Comment by pedja1** [comment]
I also have the same problem. nslookup from within nzbget docker container works fine, `Could not resolve hostname` on specific nzbs. Switching to other nzb, download starts fine, resuming that one specific nzb, again `Could not resolve hostname`

**Reply by dnzbk** [reply]
Interesting. Please email the failing NZB file to me at <denis@nzbget.com> so I can reproduce the issue.

**Reply by pedja1** [reply]
sent
